extensions/azure/assetindex-cosmos/src/main/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetQueryBuilder.java

```diff
@@ -0,0 +1,71 @@
+package org.eclipse.dataspaceconnector.assetindex.azure;
+
+import com.azure.cosmos.models.SqlParameter;
+import com.azure.cosmos.models.SqlQuerySpec;
+import org.eclipse.dataspaceconnector.assetindex.azure.model.AssetDocument;
+import org.eclipse.dataspaceconnector.cosmos.azure.CosmosDocument;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.asset.Criterion;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CosmosAssetQueryBuilder {
+
+    private static final String ASSET_FIELD = "wrappedInstance";
+    private static final String PROPERTIES_FIELD = "properties";
+    private static final String PATH_TO_PROPERTIES = String.join(".", AssetDocument.class.getSimpleName(), ASSET_FIELD, PROPERTIES_FIELD);
+
+    public CosmosAssetQueryBuilder() {
+        // pre-check fields of the AssetDocument class that will be used to build the queries
+        assertClassContainsField(CosmosDocument.class, ASSET_FIELD);
+        assertClassContainsField(Asset.class, PROPERTIES_FIELD);
+    }
+
+    public SqlQuerySpec from(AssetSelectorExpression expression) {
```

Comment:
could this just return a String? Then we keep surfacing the Cosmos DB SDK to a minimum